pkg/cmd/checkendpoints/controller/updates_manager.go

```diff
@@ -0,0 +1,75 @@
+package controller
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoints/operatorcontrolplane/podnetworkconnectivitycheck/v1alpha1helpers"
+)
+
+// NewUpdatesManager returns a queue sorting UpdateManager with the specified sorting window.
+func NewUpdatesManager(window time.Duration) *updatesManager {
+	return &updatesManager{
+		window:       window,
+		sortingQueue: map[time.Time][]v1alpha1helpers.UpdateStatusFunc{},
+	}
+}
+
+// UpdatesManager manages a queue of updates. The lock must be obtained before
+// invoking any of the methods.
+type UpdatesManager interface {
+	sync.Locker
+	// Add an update to the queue.
+	Add(timestamp time.Time, updates ...v1alpha1helpers.UpdateStatusFunc)
+	// Return the contents of the queue.
+	Queue() []v1alpha1helpers.UpdateStatusFunc
+	// Delete the contents of the queue.
+	Clear()
+}
+
+// updateManage implements an UpdateManager sorts the incoming updates and queues updates
+// outside of a sorting window, anchored on one end by the latest update, for processing.
+type updatesManager struct {
+	sync.Mutex
+	window          time.Duration
```

Comment:
you need to doc these.  They aren't obvious.

Reply:
Done.